lib/auth/api.go

```diff
@@ -828,6 +841,24 @@ type OktaAccessPoint interface {
 	// DeleteUser deletes the given user from the cluster
 	DeleteUser(ctx context.Context, user string) error
 
+	// UpsertAccessList creates or updates an access list resource.
+	UpsertAccessList(context.Context, *accesslist.AccessList) (*accesslist.AccessList, error)
+
+	// DeleteAccessList removes the specified access list resource.
+	DeleteAccessList(context.Context, string) error
+
+	// UpsertAccessListMember creates or updates an access list member resource.
+	UpsertAccessListMember(ctx context.Context, member *accesslist.AccessListMember) (*accesslist.AccessListMember, error)
+
+	// DeleteAccessListMember hard deletes the specified access list member resource.
+	DeleteAccessListMember(ctx context.Context, accessList string, memberName string) error
+
+	// UpsertRole creates or updates role.
+	UpsertRole(ctx context.Context, role types.Role) (types.Role, error)
+
+	// DeleteRole deletes role by name.
+	DeleteRole(ctx context.Context, name string) error
```

Comment:
I can't help but notice that we are making a rather large interface larger, with a matching wrapper/delegate boilerplate below. This is likely just following suite on existing code, but it's also likely we could avoid both the large interface and the boilerplate with a different design.

Here's an example from e/ where we have a few small interfaces instead of a big one, with no wrappers. `auth.Server` implements it all, but we don't about that in this layer:

* https://github.com/gravitational/teleport.e/blob/623d02ab9ff8ee23b6b085e5b5dc23118e7e0fe9/lib/devicetrust/devicetrustv1/service.go#L123
* https://github.com/gravitational/teleport.e/blob/623d02ab9ff8ee23b6b085e5b5dc23118e7e0fe9/lib/devicetrust/devicetrustv1/service.go#L125

(No changes expected here, although I'd be happy to review it.)